### PR TITLE
For #7543: Add top site from home screen

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt [hunk 1]
```diff
@@ -343,8 +343,22 @@ class DefaultSessionControlController(
...
+                return
+            }
+            SupportUtils.POCKET_TRENDING_URL -> {
+                metrics.track(Event.PocketTopSiteClicked)
+            }
+        }
+
         metrics.track(Event.TopSiteOpenInNewTab)
-        if (url == SupportUtils.POCKET_TRENDING_URL) { metrics.track(Event.PocketTopSiteClicked) }
+        if (url == SupportUtils.POCKET_TRENDING_URL) {
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt:359]: We could move this to the above when (url) mock to neatly handle all these url cases.

Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt:359]: @gabrielluong Thanks for noticing, I have already moved this in the when, just forgot to remove the original tracking.

**Comment by ekager** [comment]
@gabrielluong do you have time to review this?

**Comment by liuche** [comment]
@topotropic now that we're past the release cutoff, is this something that we want to do? Looks like gl flagged this for revisiting later.

**Comment by gabrielluong** [comment]
To clean up our PR backlog, I am closing this PR for now and we can revive it when we get more directions around this feature.

**Pull request closed without merging**